packages/reporter/src/controllers/TelepathyReporterController.ts

```diff
@@ -0,0 +1,90 @@
+import axios from "axios"
+import { hexToNumber, Chain } from "viem"
+import { privateKeyToAccount } from "viem/accounts"
+import winston from "winston"
+import "dotenv/config"
+
+import lightClientContractABI from "../ABIs/TelepathyContractABI.json"
+import adapterContractABI from "../ABIs/TelepathyAdapterABI.json"
+import Multiclient from "../MultiClient"
+import { ControllerConfig } from "../utils/type"
+import settings from "../utils/settings.json"
+
+class TelepathyReporterController {
+  sourceChain: Chain
+  destinationChains: Chain[]
+  isEnabled: boolean = false
+  logger: winston.Logger
+  multiClient: Multiclient
+  reporterAddr: string
+  adapterAddr: { [chainName: string]: string }
+  constructor(props: ControllerConfig) {
+    this.sourceChain = props.sourceChain
+    this.destinationChains = props.destinationChains
+    this.isEnabled = props.isEnabled
+    this.logger = props.logger
+    this.multiClient = props.multiClient
+    this.reporterAddr = props.reporterAddress
+    this.adapterAddr = props.adapterAddress
+  }
+  async onBlocks(blockNumbers: string[]) {
+    try {
+      // Telepathy on support light client on Gnosis at the moment
+
+      for (const chain of this.destinationChains) {
+        const client = this.multiClient.getClientByChain(chain)
+        const account = privateKeyToAccount(process.env.PRIVATE_KEY as `0x${string}`)
```

Comment:
please read this :)

https://viem.sh/docs/clients/wallet.html#optional-extend-with-public-actions